mongodbatlas/resource_mongodbatlas_project_api_key.go

```diff
@@ -24,8 +24,9 @@ func resourceMongoDBAtlasProjectAPIKey() *schema.Resource {
 		},
 		Schema: map[string]*schema.Schema{
 			"project_id": {
-				Type:     schema.TypeString,
-				Optional: true,
+				Type:       schema.TypeString,
+				Optional:   true,
+				Deprecated: fmt.Sprintf(DeprecationByVersionMessageParameter, "1.16.0"),
```

Comment:
Do we have to update the data source as well?

Reply:
We should update our acceptance tests to not use the deprecated field

Reply:
> Do we have to update the data source as well?

Data source implementation remains the same. This is because the minimum attributes required to obtain information of the api key at the moment is using an api_key_id together with a project_id:
```
data "mongodbatlas_project_api_key" "test" {
  api_key_id = vars.api_key_id
  project_id = vars.project_id
}
```
(this is fetching the information of the api key, with all its project assignments)

> We should update our acceptance tests to not use the deprecated field

Yes can adjust so that eventually when field is removed less changes are required. Will however leave one test with old definition to ensure no breaking changes.